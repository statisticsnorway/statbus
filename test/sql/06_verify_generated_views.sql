
\d+ public.sector_ordered
\d+ public.sector_available
\d+ public.legal_form_ordered
\d+ public.legal_form_available
\d+ public.reorg_type_ordered
\d+ public.reorg_type_available
\d+ public.foreign_participation_ordered
\d+ public.foreign_participation_available
\d+ public.data_source_ordered
\d+ public.data_source_available
\d+ public.unit_size_ordered
\d+ public.unit_size_available
\d+ public.person_type_ordered
\d+ public.person_type_available
\d+ public.enterprise_group_type_ordered
\d+ public.enterprise_group_type_available
\d+ public.enterprise_group_role_ordered
\d+ public.enterprise_group_role_available


BEGIN;

SELECT admin.drop_table_views_for_batch_api('public.sector');

SELECT admin.drop_table_views_for_batch_api('public.legal_form');

SELECT admin.drop_table_views_for_batch_api('public.reorg_type');

SELECT admin.drop_table_views_for_batch_api('public.foreign_participation');

SELECT admin.drop_table_views_for_batch_api('public.data_source');

SELECT admin.drop_table_views_for_batch_api('public.unit_size');

SELECT admin.drop_table_views_for_batch_api('public.person_type');

SELECT admin.drop_table_views_for_batch_api('public.enterprise_group_type');

SELECT admin.drop_table_views_for_batch_api('public.enterprise_group_role');

ROLLBACK;
