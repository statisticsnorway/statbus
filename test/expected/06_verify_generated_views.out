\d+ public.sector_ordered
                                   View "public.sector_ordered"
   Column    |           Type           | Collation | Nullable | Default | Storage  | Description 
-------------+--------------------------+-----------+----------+---------+----------+-------------
 id          | integer                  |           |          |         | plain    | 
 path        | ltree                    |           |          |         | extended | 
 parent_id   | integer                  |           |          |         | plain    | 
 label       | character varying        |           |          |         | extended | 
 code        | character varying        |           |          |         | extended | 
 name        | text                     |           |          |         | extended | 
 description | text                     |           |          |         | extended | 
 active      | boolean                  |           |          |         | plain    | 
 custom      | boolean                  |           |          |         | plain    | 
 updated_at  | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT sector.id,
    sector.path,
    sector.parent_id,
    sector.label,
    sector.code,
    sector.name,
    sector.description,
    sector.active,
    sector.custom,
    sector.updated_at
   FROM sector
  ORDER BY sector.path;
Options: security_invoker=on

\d+ public.sector_available
                                  View "public.sector_available"
   Column    |           Type           | Collation | Nullable | Default | Storage  | Description 
-------------+--------------------------+-----------+----------+---------+----------+-------------
 id          | integer                  |           |          |         | plain    | 
 path        | ltree                    |           |          |         | extended | 
 parent_id   | integer                  |           |          |         | plain    | 
 label       | character varying        |           |          |         | extended | 
 code        | character varying        |           |          |         | extended | 
 name        | text                     |           |          |         | extended | 
 description | text                     |           |          |         | extended | 
 active      | boolean                  |           |          |         | plain    | 
 custom      | boolean                  |           |          |         | plain    | 
 updated_at  | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT sector_ordered.id,
    sector_ordered.path,
    sector_ordered.parent_id,
    sector_ordered.label,
    sector_ordered.code,
    sector_ordered.name,
    sector_ordered.description,
    sector_ordered.active,
    sector_ordered.custom,
    sector_ordered.updated_at
   FROM sector_ordered
  WHERE sector_ordered.active;
Options: security_invoker=on

\d+ public.legal_form_ordered
                                View "public.legal_form_ordered"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT legal_form.id,
    legal_form.code,
    legal_form.name,
    legal_form.active,
    legal_form.custom,
    legal_form.updated_at
   FROM legal_form
  ORDER BY legal_form.code;
Options: security_invoker=on

\d+ public.legal_form_available
                               View "public.legal_form_available"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT legal_form_ordered.id,
    legal_form_ordered.code,
    legal_form_ordered.name,
    legal_form_ordered.active,
    legal_form_ordered.custom,
    legal_form_ordered.updated_at
   FROM legal_form_ordered
  WHERE legal_form_ordered.active;
Options: security_invoker=on

\d+ public.reorg_type_ordered
                                 View "public.reorg_type_ordered"
   Column    |           Type           | Collation | Nullable | Default | Storage  | Description 
-------------+--------------------------+-----------+----------+---------+----------+-------------
 id          | integer                  |           |          |         | plain    | 
 code        | text                     |           |          |         | extended | 
 name        | text                     |           |          |         | extended | 
 description | text                     |           |          |         | extended | 
 active      | boolean                  |           |          |         | plain    | 
 custom      | boolean                  |           |          |         | plain    | 
 updated_at  | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT reorg_type.id,
    reorg_type.code,
    reorg_type.name,
    reorg_type.description,
    reorg_type.active,
    reorg_type.custom,
    reorg_type.updated_at
   FROM reorg_type
  ORDER BY reorg_type.code;
Options: security_invoker=on

\d+ public.reorg_type_available
                                View "public.reorg_type_available"
   Column    |           Type           | Collation | Nullable | Default | Storage  | Description 
-------------+--------------------------+-----------+----------+---------+----------+-------------
 id          | integer                  |           |          |         | plain    | 
 code        | text                     |           |          |         | extended | 
 name        | text                     |           |          |         | extended | 
 description | text                     |           |          |         | extended | 
 active      | boolean                  |           |          |         | plain    | 
 custom      | boolean                  |           |          |         | plain    | 
 updated_at  | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT reorg_type_ordered.id,
    reorg_type_ordered.code,
    reorg_type_ordered.name,
    reorg_type_ordered.description,
    reorg_type_ordered.active,
    reorg_type_ordered.custom,
    reorg_type_ordered.updated_at
   FROM reorg_type_ordered
  WHERE reorg_type_ordered.active;
Options: security_invoker=on

\d+ public.foreign_participation_ordered
                           View "public.foreign_participation_ordered"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT foreign_participation.id,
    foreign_participation.code,
    foreign_participation.name,
    foreign_participation.active,
    foreign_participation.custom,
    foreign_participation.updated_at
   FROM foreign_participation
  ORDER BY foreign_participation.code;
Options: security_invoker=on

\d+ public.foreign_participation_available
                          View "public.foreign_participation_available"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT foreign_participation_ordered.id,
    foreign_participation_ordered.code,
    foreign_participation_ordered.name,
    foreign_participation_ordered.active,
    foreign_participation_ordered.custom,
    foreign_participation_ordered.updated_at
   FROM foreign_participation_ordered
  WHERE foreign_participation_ordered.active;
Options: security_invoker=on

\d+ public.data_source_ordered
                                View "public.data_source_ordered"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT data_source.id,
    data_source.code,
    data_source.name,
    data_source.active,
    data_source.custom,
    data_source.updated_at
   FROM data_source
  ORDER BY data_source.code;
Options: security_invoker=on

\d+ public.data_source_available
                               View "public.data_source_available"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT data_source_ordered.id,
    data_source_ordered.code,
    data_source_ordered.name,
    data_source_ordered.active,
    data_source_ordered.custom,
    data_source_ordered.updated_at
   FROM data_source_ordered
  WHERE data_source_ordered.active;
Options: security_invoker=on

\d+ public.unit_size_ordered
                                 View "public.unit_size_ordered"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT unit_size.id,
    unit_size.code,
    unit_size.name,
    unit_size.active,
    unit_size.custom,
    unit_size.updated_at
   FROM unit_size
  ORDER BY unit_size.code;
Options: security_invoker=on

\d+ public.unit_size_available
                                View "public.unit_size_available"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT unit_size_ordered.id,
    unit_size_ordered.code,
    unit_size_ordered.name,
    unit_size_ordered.active,
    unit_size_ordered.custom,
    unit_size_ordered.updated_at
   FROM unit_size_ordered
  WHERE unit_size_ordered.active;
Options: security_invoker=on

\d+ public.person_role_ordered
                                View "public.person_role_ordered"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT person_role.id,
    person_role.code,
    person_role.name,
    person_role.active,
    person_role.custom,
    person_role.updated_at
   FROM person_role
  ORDER BY person_role.code;
Options: security_invoker=on

\d+ public.person_role_available
                               View "public.person_role_available"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT person_role_ordered.id,
    person_role_ordered.code,
    person_role_ordered.name,
    person_role_ordered.active,
    person_role_ordered.custom,
    person_role_ordered.updated_at
   FROM person_role_ordered
  WHERE person_role_ordered.active;
Options: security_invoker=on

\d+ public.enterprise_group_type_ordered
                           View "public.enterprise_group_type_ordered"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT enterprise_group_type.id,
    enterprise_group_type.code,
    enterprise_group_type.name,
    enterprise_group_type.active,
    enterprise_group_type.custom,
    enterprise_group_type.updated_at
   FROM enterprise_group_type
  ORDER BY enterprise_group_type.code;
Options: security_invoker=on

\d+ public.enterprise_group_type_available
                          View "public.enterprise_group_type_available"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT enterprise_group_type_ordered.id,
    enterprise_group_type_ordered.code,
    enterprise_group_type_ordered.name,
    enterprise_group_type_ordered.active,
    enterprise_group_type_ordered.custom,
    enterprise_group_type_ordered.updated_at
   FROM enterprise_group_type_ordered
  WHERE enterprise_group_type_ordered.active;
Options: security_invoker=on

\d+ public.enterprise_group_role_ordered
                           View "public.enterprise_group_role_ordered"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT enterprise_group_role.id,
    enterprise_group_role.code,
    enterprise_group_role.name,
    enterprise_group_role.active,
    enterprise_group_role.custom,
    enterprise_group_role.updated_at
   FROM enterprise_group_role
  ORDER BY enterprise_group_role.code;
Options: security_invoker=on

\d+ public.enterprise_group_role_available
                          View "public.enterprise_group_role_available"
   Column   |           Type           | Collation | Nullable | Default | Storage  | Description 
------------+--------------------------+-----------+----------+---------+----------+-------------
 id         | integer                  |           |          |         | plain    | 
 code       | text                     |           |          |         | extended | 
 name       | text                     |           |          |         | extended | 
 active     | boolean                  |           |          |         | plain    | 
 custom     | boolean                  |           |          |         | plain    | 
 updated_at | timestamp with time zone |           |          |         | plain    | 
View definition:
 SELECT enterprise_group_role_ordered.id,
    enterprise_group_role_ordered.code,
    enterprise_group_role_ordered.name,
    enterprise_group_role_ordered.active,
    enterprise_group_role_ordered.custom,
    enterprise_group_role_ordered.updated_at
   FROM enterprise_group_role_ordered
  WHERE enterprise_group_role_ordered.active;
Options: security_invoker=on

BEGIN;
SELECT admin.drop_table_views_for_batch_api('public.sector');
 drop_table_views_for_batch_api 
--------------------------------
 
(1 row)

SELECT admin.drop_table_views_for_batch_api('public.legal_form');
 drop_table_views_for_batch_api 
--------------------------------
 
(1 row)

SELECT admin.drop_table_views_for_batch_api('public.reorg_type');
 drop_table_views_for_batch_api 
--------------------------------
 
(1 row)

SELECT admin.drop_table_views_for_batch_api('public.foreign_participation');
 drop_table_views_for_batch_api 
--------------------------------
 
(1 row)

SELECT admin.drop_table_views_for_batch_api('public.data_source');
 drop_table_views_for_batch_api 
--------------------------------
 
(1 row)

SELECT admin.drop_table_views_for_batch_api('public.unit_size');
 drop_table_views_for_batch_api 
--------------------------------
 
(1 row)

SELECT admin.drop_table_views_for_batch_api('public.person_role');
 drop_table_views_for_batch_api 
--------------------------------
 
(1 row)

SELECT admin.drop_table_views_for_batch_api('public.enterprise_group_type');
 drop_table_views_for_batch_api 
--------------------------------
 
(1 row)

SELECT admin.drop_table_views_for_batch_api('public.enterprise_group_role');
 drop_table_views_for_batch_api 
--------------------------------
 
(1 row)

ROLLBACK;
